webgl-renderers/WebGLRenderer.js

```diff
@@ -134,9 +134,92 @@ function WebGLRenderer(canvas, compositor) {
     this.bufferRegistry.allocate(cutout.spec.id, 'a_texCoord', cutout.spec.bufferValues[1], 2);
     this.bufferRegistry.allocate(cutout.spec.id, 'a_normals', cutout.spec.bufferValues[2], 3);
     this.bufferRegistry.allocate(cutout.spec.id, 'indices', cutout.spec.bufferValues[3], 1);
+
+    /**
+     * WebGL Picking for hit testing
+     */
+    this.listeners = [];
+    this.meshIds = 0;
+    this.canvas.onmousedown = this.handleClick.bind(this);
 }
 
 /**
+ * Handle mousedown clicks on Canvas for determining the position of the cursor, in order to do WebGL picking.
+ *
+ * @method
+ *
+ * @param {Object} ev Event object
+ *
+ * @return {undefined} undefined
+ */
+WebGLRenderer.prototype.handleClick = function handleClicke(ev) {
+    var x = ev.clientX;
+    var y = ev.clientY;
+
+    this.pixelRatio = window.devicePixelRatio || 1;
```

Comment:
Can we determine the pixelRatio once in the constructor?

1. It dynamically adds a new property
2. It's redundant since you're determining the pixelRatio on every click and every size change.